wstool/MemoryTool.cpp

```diff
@@ -0,0 +1,261 @@
+}
+
+int main(int argc, char **argv)
+{
+    // writeAll is a flag to indicate if working set size for every moment should be monitored
+    // or we only need to know the maximum working set size.
+    bool writeAll = false;
+    cl::ParseCommandLineOptions(argc, argv);
+    ProcessTrace(inputTrace, &WorkingSet::Process, "working set analysis", false);
+
+    int64_t internalSampleTime = 10000;
+    if (internalSampleTime > WorkingSet::timing)
+    {
+        internalSampleTime = WorkingSet::timing;
+    }
+    vector<string> InputkeyVector;    // InputkeyVector is a vector which stores keys of input virtual address map
+    vector<string> InternelkeyVector; // InternelkeyVector is a vector which stores keys of internal virtual address map
+    // InternelkeyVector is a vector which stores keys of a part of internal virtual address map
+    vector<string> InternelSampleKeyVector;
+    map<string, vector<int64_t>> InputVirAddr;
+    map<string, vector<int64_t>> InternelVirAddr;
+    map<string, vector<int64_t>> InternelSampleVirAddr;
+
+    // devide the virtual address map into two, one is internal map whose start time is bigger than 1
+    // another is input map whose start time is -1
+    //for (map<string, vector<int64_t>>::iterator it = WorkingSet::virAddr.begin(); it != WorkingSet::virAddr.end(); ++it)
+    for (auto &it : WorkingSet::virAddr)
+    {
+        if ((it.second)[1] > 0)
+        {
+            InternelkeyVector.push_back(it.first);
+            InternelVirAddr.insert(pair<string, vector<int64_t>>(it.first, it.second));
+            if ((it.second)[1] < internalSampleTime)
+            {
+                InternelSampleKeyVector.push_back(it.first);
+                InternelSampleVirAddr.insert(pair<string, vector<int64_t>>(it.first, it.second));
+            }
+        }
+        else
+        {
+            InputkeyVector.push_back(it.first);
+            InputVirAddr.insert(pair<string, vector<int64_t>>(it.first, it.second));
+        }
+    }
+    if (writeAll)
+    {
+        // for the internal map, sort the key vector according to the start time
+        std::sort(InternelkeyVector.begin(), InternelkeyVector.end(), sortKeyInternel);
+        // for the input map, sort the key vector according to the first load time
+        std::sort(InputkeyVector.begin(), InputkeyVector.end(), sortKeyInput);
+    }
+    else
+    {
+        std::sort(InternelSampleKeyVector.begin(), InternelSampleKeyVector.end(), sortKeyInternel);
+    }
+
+    uint64_t alivenumber = WorkingSet::inputSize;
+    vector<uint64_t> outputWS;
+    vector<uint64_t> inputWS;
+    vector<uint64_t> internalWS;
+    uint64_t maxInput = 0;
+    uint64_t maxOutput = 0;
+    uint64_t maxinternal = 0;
+    vector<string> outputList;
+
+    if (writeAll)
+    {
+        for (int64_t time = 0; time < WorkingSet::timing; time++)
+        {
+            vector<string> inputList;
+            vector<string> timeline;
```

Comment:
You need to really not use strings to represent these times. These need to be a uint64